### ISSUE TITLE
[asan] test_treq leaks

### ISSUE THREAD

**Issue by trws** [flux-framework/flux-core#1731]
```
=================================================================
==31001==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 40 byte(s) in 1 object(s) allocated from:
    #0 0x7f7357ced0b8 in calloc (/usr/lib/x86_64-linux-gnu/libasan.so.5+0xe90b8)
    #1 0x561306a84df7 in treq_create_common ../../../../src/modules/kvs/treq.c:207
    #2 0x561306a855eb in treq_create_rank ../../../../src/modules/kvs/treq.c:256
    #3 0x561306a83e32 in treq_basic_tests ../../../../src/modules/kvs/test/treq.c:115
    #4 0x561306a834f6 in main ../../../../src/modules/kvs/test/treq.c:353
    #5 0x7f7356fe9b16 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x22b16)

Direct leak of 40 byte(s) in 1 object(s) allocated from:
    #0 0x7f7357ced0b8 in calloc (/usr/lib/x86_64-linux-gnu/libasan.so.5+0xe90b8)
    #1 0x561306a84df7 in treq_create_common ../../../../src/modules/kvs/treq.c:207
    #2 0x561306a85537 in treq_create ../../../../src/modules/kvs/treq.c:234
    #3 0x561306a8387e in treq_add_error_cb ../../../../src/modules/kvs/test/treq.c:291
    #4 0x561306a853dc in treq_mgr_iter_transactions ../../../../src/modules/kvs/treq.c:133
    #5 0x561306a84b0c in treq_mgr_iter_tests ../../../../src/modules/kvs/test/treq.c:329
    #6 0x561306a8350a in main ../../../../src/modules/kvs/test/treq.c:357
    #7 0x7f7356fe9b16 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x22b16)

Indirect leak of 64 byte(s) in 1 object(s) allocated from:
    #0 0x7f7357ceced0 in malloc (/usr/lib/x86_64-linux-gnu/libasan.so.5+0xe8ed0)
    #1 0x7f7357677e3a in json_array (/usr/lib/x86_64-linux-gnu/libjansson.so.4+0x8e3a)
    #2 0x561306a85537 in treq_create ../../../../src/modules/kvs/treq.c:234
    #3 0x561306a8387e in treq_add_error_cb ../../../../src/modules/kvs/test/treq.c:291
    #4 0x561306a853dc in treq_mgr_iter_transactions ../../../../src/modules/kvs/treq.c:133
    #5 0x561306a84b0c in treq_mgr_iter_tests ../../../../src/modules/kvs/test/treq.c:329
    #6 0x561306a8350a in main ../../../../src/modules/kvs/test/treq.c:357
    #7 0x7f7356fe9b16 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x22b16)

Indirect leak of 64 byte(s) in 1 object(s) allocated from:
    #0 0x7f7357ceced0 in malloc (/usr/lib/x86_64-linux-gnu/libasan.so.5+0xe8ed0)
    #1 0x7f7357677e3a in json_array (/usr/lib/x86_64-linux-gnu/libjansson.so.4+0x8e3a)
    #2 0x561306a855eb in treq_create_rank ../../../../src/modules/kvs/treq.c:256
    #3 0x561306a83e32 in treq_basic_tests ../../../../src/modules/kvs/test/treq.c:115
    #4 0x561306a834f6 in main ../../../../src/modules/kvs/test/treq.c:353
    #5 0x7f7356fe9b16 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x22b16)

Indirect leak of 48 byte(s) in 1 object(s) allocated from:
    #0 0x7f7357ced0b8 in calloc (/usr/lib/x86_64-linux-gnu/libasan.so.5+0xe90b8)
    #1 0x7f73573fcd72 in zlist_new (/usr/lib/x86_64-linux-gnu/libczmq.so.4+0x28d72)
    #2 0x561306a855eb in treq_create_rank ../../../../src/modules/kvs/treq.c:256
    #3 0x561306a83e32 in treq_basic_tests ../../../../src/modules/kvs/test/treq.c:115
    #4 0x561306a834f6 in main ../../../../src/modules/kvs/test/treq.c:353
    #5 0x7f7356fe9b16 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x22b16)

Indirect leak of 48 byte(s) in 1 object(s) allocated from:
    #0 0x7f7357ced0b8 in calloc (/usr/lib/x86_64-linux-gnu/libasan.so.5+0xe90b8)
    #1 0x7f73573fcd72 in zlist_new (/usr/lib/x86_64-linux-gnu/libczmq.so.4+0x28d72)
    #2 0x561306a85537 in treq_create ../../../../src/modules/kvs/treq.c:234
    #3 0x561306a8387e in treq_add_error_cb ../../../../src/modules/kvs/test/treq.c:291
    #4 0x561306a853dc in treq_mgr_iter_transactions ../../../../src/modules/kvs/treq.c:133
    #5 0x561306a84b0c in treq_mgr_iter_tests ../../../../src/modules/kvs/test/treq.c:329
    #6 0x561306a8350a in main ../../../../src/modules/kvs/test/treq.c:357
    #7 0x7f7356fe9b16 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x22b16)

Indirect leak of 40 byte(s) in 1 object(s) allocated from:
    #0 0x7f7357ceced0 in malloc (/usr/lib/x86_64-linux-gnu/libasan.so.5+0xe8ed0)
    #1 0x7f7357677e0a in json_array (/usr/lib/x86_64-linux-gnu/libjansson.so.4+0x8e0a)
    #2 0x561306a85537 in treq_create ../../../../src/modules/kvs/treq.c:234
    #3 0x561306a8387e in treq_add_error_cb ../../../../src/modules/kvs/test/treq.c:291
    #4 0x561306a853dc in treq_mgr_iter_transactions ../../../../src/modules/kvs/treq.c:133
    #5 0x561306a84b0c in treq_mgr_iter_tests ../../../../src/modules/kvs/test/treq.c:329
    #6 0x561306a8350a in main ../../../../src/modules/kvs/test/treq.c:357
    #7 0x7f7356fe9b16 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x22b16)

Indirect leak of 40 byte(s) in 1 object(s) allocated from:
    #0 0x7f7357ceced0 in malloc (/usr/lib/x86_64-linux-gnu/libasan.so.5+0xe8ed0)
    #1 0x7f7357677e0a in json_array (/usr/lib/x86_64-linux-gnu/libjansson.so.4+0x8e0a)
    #2 0x561306a855eb in treq_create_rank ../../../../src/modules/kvs/treq.c:256
    #3 0x561306a83e32 in treq_basic_tests ../../../../src/modules/kvs/test/treq.c:115
    #4 0x561306a834f6 in main ../../../../src/modules/kvs/test/treq.c:353
    #5 0x7f7356fe9b16 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x22b16)

Indirect leak of 14 byte(s) in 1 object(s) allocated from:
    #0 0x7f7357ceced0 in malloc (/usr/lib/x86_64-linux-gnu/libasan.so.5+0xe8ed0)
    #1 0x7f735703e777 in vasprintf (/lib/x86_64-linux-gnu/libc.so.6+0x77777)

Indirect leak of 7 byte(s) in 1 object(s) allocated from:
    #0 0x7f7357c3e7e0 in strdup (/usr/lib/x86_64-linux-gnu/libasan.so.5+0x3a7e0)
    #1 0x561306a85547 in treq_create ../../../../src/modules/kvs/treq.c:239
    #2 0x561306a8387e in treq_add_error_cb ../../../../src/modules/kvs/test/treq.c:291
    #3 0x561306a853dc in treq_mgr_iter_transactions ../../../../src/modules/kvs/treq.c:133
    #4 0x561306a84b0c in treq_mgr_iter_tests ../../../../src/modules/kvs/test/treq.c:329
    #5 0x561306a8350a in main ../../../../src/modules/kvs/test/treq.c:357
    #6 0x7f7356fe9b16 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x22b16)

SUMMARY: AddressSanitizer: 405 byte(s) leaked in 10 allocation(s).
ERROR: test_treq.t - missing test plan
ERROR: test_treq.t - exited with status 1
```

**Comment by chu11** [comment]
looks like just some lazy non-cleanups in tests

**Comment by chu11** [comment]
closing, fixed in commit 17e696c767aff9bddc03e74fc07814ccfba13d5d, I put the wrong issue number in the commit message.  Doh
